lib/system/seqs_v2.nim

```diff
@@ -73,11 +73,10 @@ proc shrink*[T](x: var seq[T]; newLen: Natural) =
   when nimvm:
     setLen(x, newLen)
   else:
-    mixin `=destroy`
     #sysAssert newLen <= x.len, "invalid newLen parameter for 'shrink'"
     when not supportsCopyMem(T):
       for i in countdown(x.len - 1, newLen):
-        `=destroy`(x[i])
+        reset x[i]
```

Comment:
This is quite useful and should make it into the manual. Custom destructors should call ``reset`` rather than ``=destroy`` if in doubt.